src/libraries/UniswapV3Lib.sol

```diff
@@ -148,6 +160,58 @@ library UniswapV3Lib {
         );
     }
 
+    function removeLiquidity(UniV3Context calldata context, RemoveLiquidityParams calldata params)
+        external
+        returns (uint256 amount0Collected, uint256 amount1Collected)
+    {
+        IUniswapV3PoolLike pool = IUniswapV3PoolLike(context.pool);
+
+        _validateRemoveLiquidityParams(pool, params);
+        require(params.positionManager.ownerOf(params.tokenId) == address(context.proxy), "UniswapV3Lib/proxy-does-not-own-token-id");
+
+        address token0 = pool.token0();
+        address token1 = pool.token1();
```

Comment:
`_validateRemoveLiquidityParams` also grabs `token0` and `token1`, so they could be returned form that to save the call